src/containers/PageBuilder/LayoutComposer/LayoutComposer.js

```diff
@@ -0,0 +1,233 @@
+  });
+  return resizeListeners;
+};
+
+// Parse default areas for state hook.
+const parseDefaultAreasFromProps = props => {
+  const { areas, responsiveAreas } = props;
+  if (areas) {
+    return parseAreas(areas);
+  } else if (responsiveAreas) {
+    const firstKey = Object.keys(responsiveAreas)[0];
+    const firstAreasString = responsiveAreas?.[firstKey]?.areas;
+    if (firstAreasString) {
+      return parseAreas(firstAreasString);
+    }
+  }
+  throw new Error(
+    'LayoutComposer needs to have either "areas" or "responsiveAreas" included into props.'
+  );
+};
+
+/**
+ * LayoutComposer creates container and area wrappers using CSS Grid Template Areas.
+ * Example:
+ *
+ * const layoutAreas = `
+ *   topbar
+ *   main
+ *   footer
+ * `;
+ *
+ * return (
+ *   <LayoutComposer areas={layoutAreas} className={css.layout}>
+ *     {props => {
+ *       const { Topbar, Main, Footer } = props;
+ *       return (
+ *         <>
+ *           <Topbar as="h1" className={css.topbar}>
+ *             Hello world!
+ *           </Topbar>
+ *           <Main as="main" className={css.main}>
+ *             Some custom content.
+ *           </Main>
+ *           <Footer>
+ *             Contact us
+ *           </Footer>
+ *         </>
+ *       );
+ *     }}
+ *   </LayoutComposer>
+ * );
+ *
+ * Note: "areas" and "responsiveAreas" are alternative props.
+ * For the "responsiveAreas", the content should look like this:
+ *  {
+ *    areasSmall: {
+ *      mediaQuery: '(max-width: 767px)',
+ *      areas: `
+ *        topbar
+ *        main
+ *        extra
+ *        footer
+ *      `,
+ *    },
+ *    areasMedium: {
+ *      mediaQuery: '(min-width: 768px) and (max-width: 1023px)',
+ *      areas: `
+ *      topbar topbar
+ *      main extra
+ *      footer footer
+ *      `,
+ *    },
+ *    areasLarge: {
+ *      mediaQuery: '(min-width: 1024px)',
+ *      areas: `
+ *      topbar topbar topbar
+ *      main extra footer
+ *      `,
+ *    },
+ * }
+ *
+ * @param {Props} props for LayoutComposer (at least: children, style, areas, display, as)
+ * @return LayoutComposer that expects children to be a function.
+ */
+const LayoutComposer = React.forwardRef((props, ref) => {
+  const [currentAreas, setAreas] = useState(parseDefaultAreasFromProps(props));
+
+  useEffect(() => {
+    let resizeListeners = [];
+    if (responsiveAreas) {
+      resizeListeners = handleResponsiveAreasOnBrowser(responsiveAreas, setAreas);
+    }
+
+    return () => {
+      resizeListeners.forEach(listener => {
+        const { mediaQueryList, resizeListener } = listener;
+        mediaQueryList.removeEventListener('change', resizeListener);
+      });
+    };
+  }, []);
+
+  const { components, gridTemplateAreas } = currentAreas;
+  const { children, style = {}, areas, responsiveAreas, display, as, ...otherProps } = props;
+  const Tag = as || 'div';
+
+  return (
+    <Tag
+      {...otherProps}
+      style={{
+        ...style,
+        gridTemplateAreas,
```

Comment:
Note to self: try to avoid off this.

Reply:
The workaround is pretty complex (it would mean writing classes/media-query syntax to the `<head>` section using a `<style>` tag.